arduino/src/arduino.cpp

```diff
@@ -0,0 +1,285 @@
+ */
+Arduino::ERROR_ARDUINO Arduino::getDataWithWhileLoop()
+{
+    if (openSerial() == ERROR_WITH_SERIALPORT)
+    {
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_SERIALPORT;
+    }
+    while (true)
+    {
+        if (this->serial.available())
+        {
+            uint32_t num = 0;
+            this->serial.readBytes(&num, sizeof(num));
+            switch (num)
+            {
+            case HEADERS::GET_FLOAT:
+                receiveDataFromSensor();
+                continue;
+            case HEADERS::ASK_FOR_OVERALL_STATUS:
+                checkStatusFromArduino();
+                continue;
+            default:
+                continue;
+            }
+            return Arduino::ERROR_ARDUINO::SUCCSESS;
+        }
+    }
+}
+/**
+ * @brief will receive a message and write it to the sensor log
+ * @return Arduino::ERROR_ARDUINO will return the error code if there is one. 
+ */
+Arduino::ERROR_ARDUINO Arduino::receiveDataFromSensor()
+{
+    uint32_t id = 0;
+    this->serial.readBytes(&id, sizeof(id));
+    float f = receiveFloat();
+    this->writeFromBufferToFile(std::to_string(f), id);
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief will receive float from serial port
+ * 
+ * @return float the value that came out.
+ */
+float Arduino::receiveFloat()
+{
+    float value = 0;
+    this->serial.readBytes(&value, sizeof(value));
+    return value;
+}
+/**
+ * @brief will add mock files for the logging
+ */
+void Arduino::addLogFilesForSensors()
+{
+    this->addFile("..//dataFromArduino//sensor0.log");
+    this->addFile("..//dataFromArduino//sensor1.log");
+    this->addFile("..//dataFromArduino//sensor2.log");
+    this->addFile("..//dataFromArduino//sensor3.log");
+    this->addFile("..//dataFromArduino//sensor4.log");
+    this->addFile("..//dataFromArduino//sensor5.log");
+}
+/**
+ * @brief the simplest way to receive a boolean from the arduino
+ * @return bool the value that came from the serial port.
+ */
+bool Arduino::receiveBoolean()
+{
+    bool value = 0;
+    this->serial.readBytes(&value, sizeof(value));
+    return value;
+}
+/**
+ * @brief the function will send the arduino a message to 
+ * start checking all of its modules. It will look for errors
+ * and if there are it will reboot the arduino.
+ * 
+ * this is a wrapper.
+ * 
+ */
+Arduino::ERROR_ARDUINO Arduino::checkStatusFromArduino()
+{
+    if (receiveBoolean())
+    {
+        std::cout << "the Status is great!" << std::endl;
+    }
+    else
+    {
+        std::cout << "the Status is bad!" << std::endl;
+        return Arduino::ERROR_ARDUINO::STATUS_IS_BAD;
+    }
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief this is a wrapper for the message checking.
+ *  the c++ language cant declare threaders from other scopes 
+ * so I found a solution to this, just wrap the function inside the
+ * scope of the arduino so it will be in this scope.
+ *
+ * I use here a lambda to make it make look prettier.
```

Comment:
dont tell me why you do it and dont tell me how you do it just tell me what you are doing

Reply:
fixed